src/DwellingHydrator.php

```diff
@@ -31,9 +31,18 @@ public function loadAllDwellings(): array
     }
     /**
      * takes database connection and joins data from tables to return one dwelling object
+     * @return array of single dwelling object
      */
-    public function loadDwelling()
-    {
-        //todo
+    public function loadSingleDwelling(): Object {
+        $query = $this->db->prepare('SELECT `dwellings`.`dwellingId`, `dwellings`.`agentRef`, `dwellings`.`address1`, `dwellings`.`address2`, `dwellings`.`town`, `dwellings`.`postcode`, `dwellings`.`description`, `dwellings`.`bedrooms`, `dwellings`.`price`, `dwellings`.`image`, `types`.`type`, `statuses`.`status`                      
+                                    FROM ((`dwellings` 
+                                    INNER JOIN `types` ON `dwellings`.`typeId` = `types`.`typeId`) 
+                                    INNER JOIN `statuses` ON `dwellings`.`statusId` = `statuses`.`statusId`)
+                                    WHERE `dwellings`.`dwellingId` = :dwellingId'
+                                    );
+        $query->bindParam(':dwellingId', $_GET['dwellingId']);
+
+        $query->execute();
+        return ($query->fetchAll(\PDO::FETCH_CLASS, 'NMF\Dwelling'))[0];
```

Comment:
You should be using fetch here to get a single result, not fetchAll, then you don't need to be accessing the first element.